electrumsv/network_support/mapi.py

```diff
@@ -172,33 +179,40 @@ async def get_fee_quote(server: "NewServer",
     selection"""
-    server_state = server.api_key_state[credential_id]
-    server_state.record_attempt()
-
-    url = server.url if server.url.endswith("/") else server.url +"/"
-    url += "feeQuote"
-    headers = { 'Content-Type': 'application/json' }
-    headers.update(server.get_authorization_headers(credential_id))
-    is_ssl = url.startswith("https")
-
-    async with aiohttp.ClientSession() as client:
-        async with client.get(url, headers=headers, ssl=is_ssl) as resp:
-            try:
-                json_response = await decode_response_body(resp)
-            except (ClientConnectorError, ConnectionError, OSError, SOCKSError):
-                logger.error("failed connecting to %s", url)
-            else:
-                if resp.status != 200:
-                    logger.error("feeQuote request to %s failed with: status: %s, reason: %s",
-                        url, resp.status, resp.reason)
+    try:
+        server.api_key_state[credential_id].record_attempt()
+
+        url = server.url if server.url.endswith("/") else server.url +"/"
+        url += "feeQuote"
+        headers = {'Content-Type': 'application/json'}
+        headers.update(server.get_authorization_headers(credential_id))
+        is_ssl = url.startswith("https")
+
+        async with aiohttp.ClientSession() as client:
+            async with client.get(url, headers=headers, ssl=is_ssl) as resp:
+                try:
+                    json_response = await decode_response_body(resp)
+                except (ClientConnectorError, ConnectionError, OSError, SOCKSError):
+                    logger.error("failed connecting to %s", url)
+                    resp.raise_for_status()
                 else:
-                    assert json_response['encoding'].lower() == 'utf-8'
+                    if resp.status != 200:
+                        # We hope that this service will become available later. Until then it
+                        # should be excluded by prioritisation/server selection algorithms
+                        logger.error("feeQuote request to %s failed with: status: %s, reason: %s",
+                            url, resp.status, resp.reason)
+                    else:
+                        assert json_response['encoding'].lower() == 'utf-8'
 
-                    fee_quote_response = cast(JSONEnvelope, json_response)
-                    validate_json_envelope(fee_quote_response)
-                    logger.debug("fee quote received from %s", server.url)
+                        fee_quote_response = cast(JSONEnvelope, json_response)
+                        validate_json_envelope(fee_quote_response)
+                        logger.debug("fee quote received from %s", server.url)
 
-                    server_state.update_fee_quote(fee_quote_response)
+                        server.api_key_state[credential_id].update_fee_quote(fee_quote_response)
+    except Exception as e:
```

Comment:
This should not be needed since you await the call. For cases where it is a problem we should perhaps have a version of spawn that wraps the passed in coroutine to catch any raised exception, and does not return a future, nor wait for completion.

Reply:
Yeah. I think it was previously swallowing the exception when using app_state.async_.spawn but I have refactored slightly to use `async/await` syntax for all of the broadcasting logic. Will remove the `try/except` here. Thanks.